### PR TITLE
[AIRFLOW-3980] Unify logger

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

  - https://issues.apache.org/jira/browse/AIRFLOW-3980

### Description

The logger was used incorrectly in several places. We should not format the text before passing it to the logger, because it can be ignored in the logger.

CC: @BasPH 

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by XD-DENG** [comment]
> ### Description
> The logger was used incorrectly in several places. We should not format the text before passing it to the logger, because it can be ignored in the logger.

Hi @mik-laj , does logger ignore format method? In which cases?

**Comment by BasPH** [comment]
The logger does know `str.format()` but it's best practice in Python to use this form with logging: `logger.info("Hello %s", "world")` since that creates a string _only_ when it actually gets logged. `str.format()` and f-strings create strings before they get passed to the logger, even if the string is never logged. So it's considered inefficient and not advised.

@mik-laj Nice work! Also got rid of some of the `**locals()`👍.

**Comment by mik-laj** [comment]
If parameters and text are pass separately, they can also be independently analyzed if more advanced loggers are used.

**Comment by mik-laj** [comment]
I can not find any more problems. Can I ask you to review and accept changes?

File: airflow/contrib/sensors/sagemaker_training_sensor.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def get_sagemaker_response(self):
             billable_time = \
                 (self.last_description['TrainingEndTime'] - self.last_description['TrainingStartTime']) * \
                 self.last_description['ResourceConfig']['InstanceCount']
-            self.log.info('Billable seconds:{}'.format(int(billable_time.total_seconds()) + 1))
+            self.log.info('Billable seconds:, %s', int(billable_time.total_seconds()) + 1)
```
Comment (Fokko) [inline airflow/contrib/sensors/sagemaker_training_sensor.py]: Maybe remove the `,` after the `:`

File: tests/utils/log/test_es_task_handler.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def test_read_raises(self):
             with mock.patch("elasticsearch_dsl.Search.execute") as mock_execute:
                 mock_execute.side_effect = Exception('Failed to read')
                 logs, metadatas = self.es_task_handler.read(self.ti, 1)
-            msg = "Could not read log with log_id: {}".format(self.LOG_ID)
+            msg = "Could not read log with log_id:"
```
Comment (Fokko) [inline tests/utils/log/test_es_task_handler.py]: Maybe just inline the `msg`?

**Review by Fokko — COMMENTED** [review]
Two comments, apart from that: LGTM.

Thanks for picking this up

File: airflow/contrib/sensors/sagemaker_training_sensor.py (hunk 1 above)
Comment (mik-laj) [reply airflow/contrib/sensors/sagemaker_training_sensor.py]: Done.

File: tests/utils/log/test_es_task_handler.py (hunk 2 above)
Comment (mik-laj) [reply tests/utils/log/test_es_task_handler.py]: Done.

**Comment by mik-laj** [comment]
I updated a PR with all suggestion.

**Comment by Fokko** [comment]
Thanks @mik-laj 

Unfortunately the CI is sad now:
```
======================================================================
33) FAIL: test_exception_propagation (tests.executors.test_celery_executor.CeleryExecutorTest)
----------------------------------------------------------------------
   Traceback (most recent call last):
    tests/executors/test_celery_executor.py line 162 in test_exception_propagation
      self.assertIn('AttributeError', args[1])
   AssertionError: 'AttributeError' not found in AttributeError("'dict' object has no attribute 'state'",)
```

**Comment by mik-laj** [comment]
@Fokko Now Travis is green.

**Comment by Fokko** [comment]
Thanks @mik-laj

**Pull request merged**